### ISSUE TITLE
CurlClient and Facebook service

### ISSUE THREAD

**Issue by sjimi** [Lusitanian/PHPoAuthLib#147]
When using the CurlClient instead of the StreamClient I get following error:

```
ErrorException
Undefined index: access_token
/vagrant/app/vendor/lusitanian/oauth/src/OAuth/OAuth2/Service/Facebook.php
        if (null === $data || !is_array($data)) {
            throw new TokenResponseException('Unable to parse response.');
        } elseif (isset($data['error'])) {
            throw new TokenResponseException('Error in retrieving token: "' . $data['error'] . '"');
        }

        $token = new StdOAuth2Token();
        $token->setAccessToken($data['access_token']);
        $token->setLifeTime($data['expires']);
```

If I tracked it correctly this is the result of the following:

```
// Facebook gives us a query string ... Oh wait. JSON is too simple, understand ?
parse_str($responseBody, $data);
```

Facebook appears to give JSON if there is an ERROR. This is not handled correctly here. The current implementation when using the CurlClient does not work for errors thus.

I'm I right or am I missing something here?

**Comment by kalan2001** [comment]
It's true.

If there is any error Facebook returns a JSON

```
string '{"error":{"message":"Error validating verification code. Please make sure your redirect_uri is identical to the one you used in the OAuth dialog request","type":"OAuthException","code":100}}' (length=190)
```

Change this
// Facebook gives us a query string ... Oh wait. JSON is too simple, understand ?
parse_str($responseBody, $data);

I think the corrent form would be: 
        $data = json_decode($responseBody, true);
        if (!$data) {
            // Facebook gives us a query string ... Oh wait. JSON is too simple, understand ?
            parse_str($responseBody, $data);
        }



Please, correct it.

**Comment by scottsb** [comment]
@Lusitanian, this issue can be closed. It's been fixed in commit #27d23a4f08.

**Comment by tezrik** [comment]
This issue has been  marked as stale because it has not had recent activity. It will be closed if no further activity occurs.
If this is still an issue for you, please try to help by debugging it further and sharing your results.
Thank you for your contributions.
